### DISCUSSION TITLE
How to reduce number of volumes after increasing -volumeSizeLimitMB

### DISCUSSION THREAD

**Discussion by Zen3515** [seaweedfs/seaweedfs#5661] [Q&A]
I'm trying to stabilize a seaweedfs system (tuning the parameters).

My problem is that the write is terribly slow compared to what it used to be, I have 241338 volumes (with volumeSizeLimitMB of 1GB). I've tried to resolve the issue by setting the volumeSizeLimitMB to be 7GB instead, but I don't know how to reduce the number of volumes that were created.

A month ago, I suffered from slow vacumm that leads volume size keep increasing resulted in all my HDD becames unwritable. (reaching the 1% threshold, I need to shut down the volume server and compact it manually). After I got it back, I reduced the volumeSizeLimitMB from 30GB to just 1GB, and things were good, for 28 days.

My use case is around 5TB write daily, and 5TB delete daily. And it took about 28 days to saturate the number of volume (it has been growing all this time)
3 Days ago, the write can't keep up (my software buffer is getting full), I suspect the number of volumes is the cause, so I increase the number of volumeSizeLimitMB to 7GB (went from 30GB --28dayspass--> 1GB --28dayspass--> 7GB where we are now). I tried to delete some of the data and vacumm and volme.deleteEmpty but deleteEmpty seems to be taking a long time to realize a new empty volume appears and thus, not a very good way to reduce the number of volumes. Right now, I diverged the write of 5TB to someplace else but it's about to become full at any moment now. So, I really need some advice on how one would reduce the number of volumes.

**Comment by Zen3515** [comment]
I ended up adding a new external HBA card, and add an equivalent amount of HDD to the server, What a nightmare!

For future reference, if you could suggest anything I'd appreciate that a lot.

**Comment by chrislusf** [comment]
Does TTLed-file fit this file access pattern?
